src/Docs/Session/Create/NotificationConfigBuilder.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yoti\Docs\Session\Create;
+
+class NotificationConfigBuilder
+{
+
+    private const RESOURCE_UPDATE = 'RESOURCE_UPDATE';
+    private const TASK_COMPLETION = 'TASK_COMPLETION';
+    private const CHECK_COMPLETION = 'CHECK_COMPLETION';
+    private const SESSION_COMPLETION = 'SESSION_COMPLETION';
+
+    /**
+     * @var string
+     */
+    private $authToken;
+
+    /**
+     * @var string
+     */
+    private $endpoint;
+
+    /**
+     * @var string[]
+     */
+    private $topics = [];
+
+    public function withAuthToken(string $authToken): self
+    {
+        $this->authToken = $authToken;
+        return $this;
+    }
+
+    public function withEndpoint(string $endpoint): self
+    {
+        $this->endpoint = $endpoint;
+        return $this;
+    }
+
+    public function forResourceUpdate(): self
+    {
+        return $this->withTopic(self::RESOURCE_UPDATE);
+    }
+
+    public function withTopic(string $topic): self
+    {
+        if (!in_array($topic, $this->topics, true)) {
```

Comment:
We don't have this check in other SDKs - do you think this is something we should do across the board?

cc @echarrod

Reply:
Do you mean not adding the topic if it's already there? Seems sensible

Reply:
And we'd also obviously want a unit test to confirm that behaviour in each SDK if we do that